src/Common/error-handler-utils.ts

```diff
@@ -0,0 +1,40 @@
+import { Error as grpcWebError } from 'grpc-web'
+import { ServiceError, status } from 'grpc'
+import XpringIlpError from '../ILP/xpring-ilp-error'
+
+/**
+ * Utility class for translating gRPC errors to native Xpring SDK errors.
+ * Though grpc-web and grpc handle errors differently (returned errors are separate classes),
+ * they both follow the same error coding rules as defined here
+ * https://github.com/grpc/grpc/blob/master/doc/statuscodes.md.  This means that this class can translate
+ * any gRPC error code into a native error.
+ */
+export default class ErrorHandlerUtils {
+  /**
+   * Handle an Error thrown from an Ilp network client call by translating it to
+   * a XpringIlpError. gRPC services follow return an error with a status code,
+   * so we need to map gRPC error status to native XpringIlpErrors.
+   * GrpcNetworkClient and GrpcNetworkClientWeb also sometimes throw a XpringIlpError,
+   * so we need to handle that case in here as well.
+   *
+   * @param error Any error returned by a network call.
+   */
+  public static handleIlpServiceError(
+    error: ServiceError | grpcWebError | XpringIlpError,
```

Comment:
How will we know which method to call?  The `Error` returned from the network client is polymorphic, so we will need to do a type check on the error type anyway.  This just does the type check within the method, instead of by the caller.  

Can you expand on what you would do here?

Reply:
Ah, I guess that makes sense. Happy with this the way it is. 

I’m generally suspicious of any sort of polymorphic parameters in JavaScript although this does look like one time that they are actually warranted.

Reply:
I'm generally suspicious of most things in JavaScript